### PR TITLE
update dependencies and postcss config

### PR DESCRIPTION
**Pull request by romainmenke**
fixes : https://github.com/argyleink/open-props/issues/233

But likely also breaks things as postcss-preset-env was updated.
I am biased but postcss-preset-env v6.7.0 is getting old and has too many unresolved bugs :)

The most important thing to verify is how all the nested CSS is transformed with this new version. (6.7.0 was not spec compliant)

### CODE REVIEW

File: postcss.config.cjs [hunk 1]
```diff
@@ -14,17 +14,16 @@ module.exports = {
     postcssPresetEnv({
       stage: 0,
       autoprefixer: false,
+      enableClientSidePolyfills: false,
       features: {
-        'logical-properties-and-values': false, 
-        'prefers-color-scheme-query': false, 
-        'gap-properties': false,
+        'cascade-layers': false,
```
Comment (romainmenke) [inline postcss.config.cjs:19]: Only works well for final products, not for libraries

File: postcss.config.cjs [hunk 2]
```diff
@@ -14,17 +14,16 @@ module.exports = {
     postcssPresetEnv({
       stage: 0,
       autoprefixer: false,
+      enableClientSidePolyfills: false,
```
Comment (romainmenke) [inline postcss.config.cjs:17]: Fairly recent plugin option that disables all plugins that need client side JS to function (`:has`, `:focus-within`, ...)

**Comment by argyleink** [comment]
confirmed fix in normalize output! thanks so much ❤️ 

i did however see a few other changes in the normalize output. I couldn't put a thumb on the cause, plugins like selector merging are involved, but just a straight unminfied comparison of normalize shows just a /:has/ selector being removed cuz of the new flag to remove js plugins (cool bool btw). so i'm a little tentative to merge this pr, but i do believe an update to latest in due 👍🏻  thanks for kickin it off

i also tried just updating cssnano, and that's enough to fix #233. what do you think of a pr for updating cssnano and then this one becomes updating preset-env?

**Comment by romainmenke** [comment]
see : https://github.com/argyleink/open-props/pull/235

> i also tried just updating cssnano, and that's enough to fix https://github.com/argyleink/open-props/issues/233. what do you think of a pr for updating cssnano and then this one becomes updating preset-env?

That would be better :)

I got a bit carried away with this change, but getting a fix out first for the weird minified result will unblock some people.

**Comment by argyleink** [comment]
think its worth updating the other postcss packages while this is updating preset-env?

**Comment by argyleink** [comment]
it's very clear there are bytes to shave off my files by inspecting and managing the output better.. just spent 45m making builds of normalize with old and new postcss/preset-env/etc.

**Comment by romainmenke** [comment]
> think its worth updating the other postcss packages while this is updating preset-env?

```
npm outdated      
Package             Current  Wanted  Latest  Location                         Depended by
ava                  3.15.0  3.15.0   4.2.0  node_modules/ava                 open-props
concurrently          6.5.1   6.5.1   7.2.1  node_modules/concurrently        open-props
microbundle          0.13.3  0.13.3  0.15.0  node_modules/microbundle         open-props
postcss-cli           8.3.1   8.3.1   9.1.0  node_modules/postcss-cli         open-props
postcss-preset-env    7.6.0   7.7.0   7.7.0  node_modules/postcss-preset-env  open-props
```

_there was a release of `postcss-preset-env` yesterday with support for `sin()`, `cos()`, ..._

`postcss-cli` can be updated safely if I compare how it's used here with the changelog : https://github.com/postcss/postcss-cli/blob/master/CHANGELOG.md#900--2021-09-24

-----

> just spent 45m making builds of normalize with old and new postcss/preset-env/etc.

Was there anything unexpected there?
Always interesting to have feedback based on real world CSS :)

File: package.json [hunk 3]
```diff
@@ -153,6 +153,6 @@
     "postcss-cli": "^8.3.1",
     "postcss-combine-duplicated-selectors": "^10.0.3",
     "postcss-import": "^14.0.2",
-    "postcss-preset-env": "6.7.x"
+    "postcss-preset-env": "^7.6.0"
```
[the remark this answers is not in the document]
Comment (romainmenke) [reply package.json:156]: There is a bug rule with selector `:where(html) {` in `normalize.min.css` that is moved down a few rules.

I can not thing of anything `postcss-preset-env` related that could cause this.

[the remark this answers is not in the document]
Comment (argyleink) [reply package.json:156]: disable postcss-clamp 👍🏻 

> I do think there is typo in the source code here..

appreciate the note! i have a refactored version i've been meaning to swap out for the button styles, i wonder if i still have an issue in there too. thanks!

> bug with lines moved down a few rules in `normalize.min.css`..

i chased that one too, not even sure how to work around it yet

**Comment by argyleink** [comment]
will give this another checkout and build pass, get back to you soon. thanks!

**Comment by equinusocio** [comment]
I think this PR is important and should get attention. Providing code from old postcss-preset-env version my result in useless and bloated code.

**Pull request closed without merging**